test/box-luatest/gh_10057_user_role_test.lua

```diff
@@ -0,0 +1,67 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new()
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.after_each(function(cg)
+    cg.server:exec(function()
+        box.schema.user.drop('testuser', {if_exists = true})
+        box.schema.role.drop('director', {if_exists = true})
+        box.schema.role.drop('manager', {if_exists = true})
+        box.schema.role.drop('staff', {if_exists = true})
+    end)
+end)
+
+g.test_user_role = function(cg)
+    cg.server:exec(function()
+        box.schema.user.create('testuser')
+        box.schema.role.create('manager')
+        box.schema.role.create('director')
+        box.schema.user.grant('testuser', 'manager')
+
+        local original_user = box.session.effective_user()
+        box.session.su('testuser')
+
+        -- Check we don't have a Lua error on box.schema.info().
+        local ok = pcall(box.schema.user.info)
+        t.assert(ok)
+
+        -- Check the user has an access to the 'manager' role.
+        t.assert_equals(#box.space._vuser.index.name:select('manager'), 1)
+
+        -- Check the user doesn't have access to the 'director' role.
+        t.assert_equals(#box.space._vuser.index.name:select('director'), 0)
+
+        box.session.su(original_user)
+    end)
+end
+
+g.test_transitive_role = function(cg)
+    cg.server:exec(function()
+        box.schema.user.create('testuser')
+        box.schema.role.create('staff')
+        box.schema.role.create('manager')
+        box.schema.role.grant('manager', 'staff')
+        box.schema.user.grant('testuser', 'manager')
+
+        local original_user = box.session.effective_user()
+        box.session.su('testuser')
+
```

Comment:
Let's check `box.schema.user.info()` here as well.

Reply:
Seems redundant, it's tested above and implicitly tested here by selecting the `manager` role.

Reply:
I agree with @drewdzzz, the test shouldn't make such assumptions about internals (i.e. that covering `select` implicitly covers `box.schema.user.info()`).
Let's check `box.schema.user.info()` here as well.